src/cargo/util/errors.rs

```diff
@@ -132,6 +132,50 @@ impl fmt::Debug for ProcessError {
 }
 
 // =============================================================================
+// Cargo test errors.
+
+/// Error when testcases fail
+pub struct CargoTestError {
+    pub desc: String,
+    pub exit: Option<ExitStatus>,
+    cause: Option<io::Error>,
+}
```

Comment:
I think it'd be best for this to actually contain the literal `Vec<ProcessError>` list internally, that way we can be sure no information is lost.